Document crash_loop_limit node configuration for 23.1 (#1131)
redpanda-data/documentation: docs/reference/node-properties.mdx

---

### crash_loop_limit

> A limit on the number of consecutive crashes (unclean shutdowns) of a node within a short duration of each other, after which the node cannot restart until its crash-tracking logic is reset. It prevents a node from getting stuck in an infinite cycle of crashes.

Reviewer: ```suggestion
A limit on the number of consecutive times a broker can crash before its crash-tracking logic is reset. This limit prevents a broker from getting stuck in an infinite cycle of crashes.
```

Reviewer: 'within a short duration' is ambiguous so I took it out.

Also, please clarify 'crash-tracking logic is reset'. Is this an automated process? Or is a user supposed to trigger it?
